lunavl/sdk/detectors/humandetector.py

```diff
@@ -0,0 +1,229 @@
+"""
+Module contains function for detection on images.
+"""
+from functools import partial
+from typing import Any, Dict, List, Literal, Optional, Union, overload
+
+from FaceEngine import (  # pylint: disable=E0611,E0401
+    Face,
+    Human,
+    FSDKErrorResult,
+)
+
+from .bodydetector import BodyDetection
+from .facedetector import FaceDetection
+from ..async_task import AsyncTask
+from ..detectors.base import (
+    ImageForDetection,
+    getArgsForCoreDetectorForImages,
+    validateBatchDetectInput,
+)
+from ..errors.exceptions import assertError
+from ..image_utils.image import VLImage
+from ..launch_options import LaunchOptions
+
+
+class HumanDetection:
+    """
+    Human detection is union of face and body detection of one human.Face and body may be None if detector did not
+    detect corresponding part of the torso
+
+    Attributes:
+        face (Optional[FaceDetection]): optional body detection
+        body (Optional[BodyDetection]): optional face detection
+        associationScore (Optional[float]): body and face association score
+    """
+
+    __slots__ = ("face", "body", "_image", "associationScore")
+
+    def __init__(
+        self,
+        coreFaceDetection: Optional[Face],
+        coreBodyDetection: Optional[Human],
+        image: VLImage,
+        score: Optional[float] = None,
+    ):
+        """
+        Init.
+
+        Args:
+            coreFaceDetection: core face detection
+            coreBodyDetection: core body detection
+            image: original image
+            score:  association score of body and face. If face or body is none score must be set to None
+        """
+        self.face = FaceDetection(coreFaceDetection, image) if coreFaceDetection else None
+        self.body = BodyDetection(coreBodyDetection, image) if coreBodyDetection else None
+        self.associationScore = score
+        self._image = image
+
+    def asDict(self) -> Dict[str, Any]:
```

Comment:
Not about object behavior, but rather representation. Client's should depend on the public attributes, not the dictionary. And its kinda hard to inherit from a such method.